src/Tools/ExternalAccess/Razor/Shared/RazorMappedSpanResult.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.Razor;
+
+internal readonly struct RazorMappedSpanResult
+{
+    public readonly string FilePath;
+
+    public readonly LinePositionSpan LinePositionSpan;
+
+    public readonly TextSpan Span;
+
+    public RazorMappedSpanResult(string filePath, LinePositionSpan linePositionSpan, TextSpan span)
+    {
+        if (string.IsNullOrEmpty(filePath))
+        {
+            throw new ArgumentException(nameof(filePath));
+        }
+
+        FilePath = filePath;
+        LinePositionSpan = linePositionSpan;
+        Span = span;
+    }
+
+    public bool IsDefault => FilePath == null;
+}
+
+internal readonly record struct RazorMappedEditoResult(string FilePath, TextChange[] TextChanges)
```

Comment:
Typo in the type name? Or do we have to carry the typo forward?

Reply:
Typo that can be fixed!!! This was carried over from the razor EA that Roslyn consumed but only in VS Code via the `IRazorMappingService` lsp contract